client/my-sites/checkout/get-thank-you-page-url/index.ts

```diff
@@ -297,13 +297,18 @@ export default function getThankYouPageUrl( {
 		( cart?.create_new_blog || signupFlowName === 'domain' ) &&
 		! isDomainOnly &&
 		urlFromCookie &&
-		receiptIdOrPlaceholder &&
-		! urlFromCookie.includes( '/start/setup-site' )
+		receiptIdOrPlaceholder
 	) {
-		clearSignupCompleteFlowName();
-		const newBlogReceiptUrl = `${ urlFromCookie }/${ receiptIdOrPlaceholder }`;
-		debug( 'new blog created, so returning', newBlogReceiptUrl );
-		return newBlogReceiptUrl;
+		if ( urlFromCookie.includes( '/setup/setup-site' ) ) {
+			debug( 'domain only signup with free site selection' );
+			return urlFromCookie;
+		}
+		if ( ! urlFromCookie.includes( '/start/setup-site' ) ) {
```

Comment:
This function is very complex and it can be easy to introduce regressions so we've covered every output of the function in unit tests (see the note at the top of the file). Could you add a unit test to prove this bug and its fix?

Reply:
will do 👍